packages/core/src/modules/proofs/formats/presentation-exchange/PresentationExchangeProofFormatService.ts

```diff
@@ -0,0 +1,309 @@
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const pexFormat = proofFormats.presentationExchange
+    if (!pexFormat) {
+      throw new AriesFrameworkError('Missing Presentation Exchange format in create proposal attachment format')
+    }
+
+    const { presentationDefinition } = pexFormat
+
+    ps?.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({ format: PRESENTATION_EXCHANGE_PRESENTATION_PROPOSAL, attachmentId })
+
+    const attachment = this.getFormatData(presentationDefinition, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async processProposal(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const proposal = attachment.getDataAsJson<PresentationDefinition>()
+    ps.validatePresentationDefinition(proposal)
+  }
+
+  public async acceptProposal(
+    agentContext: AgentContext,
+    { attachmentId, proposalAttachment }: ProofFormatAcceptProposalOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const presentationDefinition = proposalAttachment.getDataAsJson<PresentationDefinition>()
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const attachment = this.getFormatData({ presentation_definition: presentationDefinition }, format.attachmentId)
+
+    return { format, attachment }
+  }
+
+  public async createRequest(
+    agentContext: AgentContext,
+    { attachmentId, proofFormats }: FormatCreateRequestOptions<PresentationExchangeProofFormat>
+  ): Promise<ProofFormatCreateReturn> {
+    const ps = this.presentationExchangeService(agentContext)
+
+    const presentationExchangeFormat = proofFormats.presentationExchange
+
+    if (!presentationExchangeFormat) {
+      throw Error('Missing presentation exchange format in create request attachment format')
+    }
+
+    const { presentationDefinition } = presentationExchangeFormat
+
+    ps.validatePresentationDefinition(presentationDefinition)
+
+    const format = new ProofFormatSpec({
+      format: PRESENTATION_EXCHANGE_PRESENTATION_REQUEST,
+      attachmentId,
+    })
+
+    const options = { challenge: 'TODO' }
+
+    const attachment = this.getFormatData(
+      { options, presentation_definition: presentationDefinition },
+      format.attachmentId
+    )
+
+    return { attachment, format }
+  }
+
+  public async processRequest(agentContext: AgentContext, { attachment }: ProofFormatProcessOptions): Promise<void> {
+    const ps = this.presentationExchangeService(agentContext)
+    const { presentation_definition: presentationDefinition } = attachment.getDataAsJson<{
```

Comment:
Same here, we should define an interface for all the attachments, and then use the interface everywhere. We need these interfaces anyway for the current `unknown` types in the format interface for proposal/request/presentation.